### PR TITLE
expression: support push down tikv supported functions

### PR DESCRIPTION
**Pull request by Yisaer**
Signed-off-by: yisaer <disxiaofei@163.com>

### What problem does this PR solve?

Issue Number: ref https://github.com/pingcap/tidb/issues/31846

### What is changed and how it works?

push following functions down to TiKV, they are already supported in TiKV accoriding to https://github.com/tikv/tikv/issues/5751

math:

* sin/asin/cos/acos/tan/atan/atan2/cot 

date: 

* to_days/hour/minute/second/microsecond/month
* dayName/dayOfMonth/dayOfWeek/dayOfYear/last_day

### Check List

Tests 

- [x] Unit test

### Release note

```release-note
None
```

### CODE REVIEW

**Review by rebelice — APPROVED** [review]

**Review by JmPotato — APPROVED** [review]

**Comment by breezewish** [comment]
Looks like this PR accidentally makes copr-test not valid for a lot of test cases, because of the addition of `failpoint.Return(false)`.

Before this PR, the copr-test check is:

- Compare tidb+mocktikv  vs   tidb+tikv+enabling_functions_[in_this_list](https://github.com/tikv/copr-test/blob/master/push-down-test/functions.txt)

After this PR, the copr-test check is:

- Compare tidb+mocktikv  vs   tidb+tikv+enabling_functions_[in_this_list](https://github.com/tikv/copr-test/blob/master/push-down-test/functions.txt)+disabling_functions_not_in_the_list

In short, if a function is not listed [in the list](https://github.com/tikv/copr-test/blob/master/push-down-test/functions.txt), it will not be tested at all (tidb+mocktikv   vs  tidb).
